src/qibo/result.py

```diff
@@ -237,6 +246,37 @@ def frequencies(self, binary=True, registers=False):
 
         return self._frequencies
 
+    def probabilities(self, qubits=None):
+        """Calculate the probabilities as frequencies / nshots
+
+        Returns:
+            The array containing the probabilities of the measured qubits.
+        """
+        nqubits = len(self.measurement_gate.qubits)
+        if qubits is None:
+            qubits = range(nqubits)
+        else:
+            if not set(qubits).issubset(self.measurement_gate.qubits):
+                raise RuntimeError(
+                    "Asking probabilities for qubits that were not measured."
+                )
+            else:
+                qubits = range(len(qubits))
+
+        if self._probs is not None:
+            return self.backend.calculate_probabilities(
+                self._probs, qubits, len(self.measurement_gate.qubits)
+            )
+
+        frequencies = self.frequencies(binary=False)
+        probs = self.backend.np.zeros(2 ** self._samples.shape[1])
+        for state, freq in frequencies.items():
+            probs[state] = freq / self.nshots
```

Comment:
This breaks in `tensorflow` because `tensorflow` does not allow item assignment on tensors